src/main/java/ca/uhn/fhir/jpa/starter/CustomConsentService.java

```diff
@@ -34,68 +37,70 @@ public CustomConsentService(DaoRegistry daoRegistry) {
   @Override
   public ConsentOutcome startOperation(RequestDetails theRequestDetails,
       IConsentContextServices theContextServices) {
-    if (!ObjectUtils.isEmpty(theRequestDetails.getHeader("Authorization"))) {
-      if (oAuth2Helper.canBeInPatientCompartment(theRequestDetails.getResourceName())) {
-        return ConsentOutcome.AUTHORIZED;
-      }
-      String patientId = getPatientFromToken(theRequestDetails);
-      String resourceName = null;
-      boolean proceed = false;
-      switch (theRequestDetails.getRequestType().toString()) {
-        case "POST":
-          resourceName = theRequestDetails.getResourceName();
+    /*
+     * Returning authorized if there is no Authorization header present or if requested resource is
+     * present in Patient Compartment For both these cases all the consent logic is in authorization
+     * intercepter rules.
+     */
+    if (ObjectUtils.isEmpty(theRequestDetails.getHeader("Authorization"))
+        || oAuth2Helper.canBeInPatientCompartment(theRequestDetails.getResourceName())) {
+      return ConsentOutcome.AUTHORIZED;
+    }
+    String patientId = getPatientFromToken(theRequestDetails);
+    String resourceName = null;
+    boolean proceed = false;
+    switch (theRequestDetails.getRequestType().toString()) {
+      case "POST":
+        resourceName = theRequestDetails.getResourceName();
+        proceed = isResourceValid(resourceName, patientId, theRequestDetails.getResource());
+        break;
+      case "PUT":
+        resourceName = theRequestDetails.getResourceName();
+        IBaseResource putResource = getResourceFromDB(theRequestDetails.getRequestPath());
```

Comment:
Minor thing, but can you rename this var to be `existingResource` since it's not the resource in the PUT request?